pkg/collector/corechecks/cluster/kubernetes_eventbundle.go

```diff
@@ -78,15 +80,44 @@ func (b *kubernetesEventBundle) addEvent(event *v1.Event) error {
 	return nil
 }
 
-func (b *kubernetesEventBundle) formatEvents(clusterName string) (metrics.Event, error) {
+func (b *kubernetesEventBundle) formatEvents(clusterName string, providerIDByNodename map[string]string) (metrics.Event, error) {
 	if len(b.events) == 0 {
 		return metrics.Event{}, errors.New("no event to export")
 	}
-	// Adding the clusterName to the nodename if present
+
+	tags := []string{fmt.Sprintf("source_component:%s", b.component), fmt.Sprintf("kubernetes_kind:%s", b.kind), fmt.Sprintf("name:%s", b.name)}
+
 	hostname := b.nodename
-	if b.nodename != "" && clusterName != "" {
-		hostname = hostname + "-" + clusterName
+	if b.nodename != "" {
+		// Adding the clusterName to the nodename if present
+		if clusterName != "" {
+			hostname = hostname + "-" + clusterName
+		}
+
+		// Find provider ID from stored map or find via node spec from APIserver
+		hostProviderID, ok := providerIDByNodename[b.nodename]
+		if !ok {
+			cl, err := as.GetAPIClient()
+			if err != nil {
+				log.Warnf("Can't create client to query the API Server: %v", err)
+			} else {
+				node, err := as.GetNode(cl, b.nodename)
+				if err != nil {
+					log.Warnf("Can't get node from API Server: %v", err)
+				} else {
+					providerID := node.Spec.ProviderID
+					if providerID != "" {
+						// e.g. gce://datadog-test-cluster/us-east1-a/some-instance-id or aws:///us-east-1e/i-instanceid
+						s := strings.Split(providerID, "/")
+						hostProviderID = s[len(s)-1]
```

Comment:
Should this new `host_provider_id:` tag match any pre-existing information somewhere ?

Because the only thing that is guaranteed to be unique is the whole provider ID.
Typically, on GCE, the `some-instance-id` will be the VM name and it is allowed to have VMs with the same name in different GCE projects (`datadog-test-cluster` in the example).
So, if a customer deploys agents targeting the same org in different GCE projects, the resulting `hostProviderID` will not be unique.

Reply:
Hm, that's tricky then. The idea was to have this tag match the instance ID we use elsewhere in the DD app. I will check if this is consistent with what we do elsewhere. Thanks for pointing it out!